src/main/java/seedu/address/model/person/PersonContainsKeywordsPredicate.java

```diff
@@ -21,7 +22,9 @@ public boolean test(Person person) {
         return keywords.stream()
                 .anyMatch(keyword -> (StringUtil.containsSubstringIgnoreCase(person.getName().fullName, keyword)
                         || StringUtil.containsSubstringIgnoreCase(person.getPhone().value, keyword)
-                                || StringUtil.containsSubstringIgnoreCase(person.getEmail().value, keyword)));
+                        || StringUtil.containsSubstringIgnoreCase(person.getEmail().value, keyword)
+                        || StringUtil.containsSubstringIgnoreCase(person.getEvents().stream().map(Event::toString)
+                        .reduce("", (x,y) -> x + " " + y), keyword)));
```

Comment:
smart way to turn all events into a string and then searching for the keyword in the string!